pulser/simulation/simulation.py

```diff
@@ -262,8 +351,8 @@ def build_coeffs_ops(basis: str, addr: str) -> list[list]:
                         # Build once global operators as they are needed
                         if op_id not in operators:
                             operators[op_id] =\
-                                self._build_operator(op_id, global_op=True)
-                        terms.append([operators[op_id], adapt(coeff)])
+                                self.build_operator([(op_id, 'global')])
+                            terms.append([operators[op_id], adapt(coeff)])
```

Comment:
Why should the append be inside the `if` now?

Reply:
I don't know I don't remember touching this part

Reply:
I think it was correct before. This way it seems like it will only work when you are building the operator, which happens only the first time.

Reply:
Did you figure out what's going on here?

Reply:
As far as I can tell, this is the only thing that needs to be changed before merging.